### PR TITLE
Fix HLS streaming (use AdaptiveMediaSource instead of MediaSource)

### PR DESCRIPTION
**Pull request by bfreydier**

### CODE REVIEW

File: samples/UniversalApplication/Converters/StreamUrlConverter.cs [hunk 1]
```diff
@@ -8,18 +8,20 @@
...
     {
         public object Convert(object value, Type targetType, object parameter, string language)
         {
-            string streamUrl = value as string;
-            if (streamUrl != null)
+            AdaptiveMediaSource mediaSource = value as AdaptiveMediaSource;
+            if (mediaSource != null)
             {
-                return MediaSource.CreateFromUri(new Uri(streamUrl));
+                return MediaSource.CreateFromAdaptiveMediaSource(mediaSource);
```
Comment (antoinecellerier) [inline samples/UniversalApplication/Converters/StreamUrlConverter.cs]: you should only do this for hls content types

File: samples/UniversalApplication/ViewModels/PlayerViewModel.cs [hunk 2]
```diff
@@ -61,7 +65,7 @@ public PlayerViewModel(IGrooveClient grooveClient, GrooveApiErrorViewModel error
 
             if (!string.IsNullOrEmpty(streamResponse.Url))
             {
-                StreamUrl = streamResponse.Url;
+                MediaSource = (await AdaptiveMediaSource.CreateFromUriAsync(new Uri(streamResponse.Url))).MediaSource;
             }
```
Comment (antoinecellerier) [inline samples/UniversalApplication/ViewModels/PlayerViewModel.cs]: It would be interesting to add a comment to explain why this is needed

**Review by antoinecellerier — CHANGES_REQUESTED** [review]
You probably need to fix non HLS content playback

File: samples/UniversalApplication/ViewModels/PlayerViewModel.cs [hunk 3]
```diff
@@ -61,7 +66,13 @@ public PlayerViewModel(IGrooveClient grooveClient, GrooveApiErrorViewModel error
 
             if (!string.IsNullOrEmpty(streamResponse.Url))
             {
-                StreamUrl = streamResponse.Url;
+                Uri streamResponseUri = new Uri(streamResponse.Url);
+
+                MediaSource =
+                    streamResponse.ContentType == "application/vnd.apple.mpegurl"
+                        ? MediaSource.CreateFromAdaptiveMediaSource(
+                            (await AdaptiveMediaSource.CreateFromUriAsync(streamResponseUri)).MediaSource)
```
Comment (daauro) [inline samples/UniversalApplication/ViewModels/PlayerViewModel.cs]: >(await AdaptiveMediaSource.CreateFromUriAsync(streamResponseUri)).MediaSource) [](start = 28, length = 78)

in the sample codes we saw this morning on the interwebs everyone seemed to check if the AdaptiveMediaSource had a success status before passing it to the MediaSource, do you know if that's important?

**Comment by daauro** [comment]
:shipit:

File: samples/UniversalApplication/ViewModels/PlayerViewModel.cs [hunk 4]
```diff
@@ -61,8 +67,28 @@ public PlayerViewModel(IGrooveClient grooveClient, GrooveApiErrorViewModel error
 
             if (!string.IsNullOrEmpty(streamResponse.Url))
             {
-                StreamUrl = streamResponse.Url;
+                MediaSource = await GetMediaSource(new Uri(streamResponse.Url), streamResponse.ContentType);
             }
         }
+
+        private async Task<MediaSource> GetMediaSource(Uri streamUri, string contentType)
```
Comment (antoinecellerier) [inline samples/UniversalApplication/ViewModels/PlayerViewModel.cs:74]: Please add a comment explaining why this is needed

**Review by antoinecellerier — APPROVED** [review]

File: samples/UniversalApplication/ViewModels/PlayerViewModel.cs (hunk 4 above)
Comment (bfreydier) [reply samples/UniversalApplication/ViewModels/PlayerViewModel.cs:74]: Added

**Pull request merged**